user-operator/src/main/java/io/strimzi/operator/user/model/KafkaUserModel.java

```diff
@@ -387,11 +395,27 @@ protected byte[] decodeFromSecret(Secret secret, String key) {
      * @return The secret.
      */
     protected Secret createSecret(Map<String, String> data) {
+        Map<String, String> labels = this.labels.toMap();
+
+        if (this.secretLabelExclusionPattern != null) {
+            LOGGER.infoOp("Applying Secret Label Exclusion Pattern: {}", this.secretLabelExclusionPattern);
+            LOGGER.debugOp("Initial labels before exclusion: {}", labels);
+
+            labels = Util.mergeLabelsOrAnnotations(labels, templateSecretLabels)
+                .entrySet().stream()
+                .filter(entry -> !this.secretLabelExclusionPattern.matcher(entry.getKey()).matches()) // Filtering based on keys
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+            LOGGER.debugOp("Updated labels: {}", labels);
+        } else {
+            LOGGER.debugOp("No Secret Label Exclusion Pattern set. Skipping label filtering.");
+        }
```

Comment:
I think this logging is a bit over the top. Maybe one debug message when it is set would be sufficient without the need to list any labels.

Reply:
Yeah, that's true. This was just for testing purposes...updating